app/controller/command/parser.py

```diff
@@ -59,27 +63,37 @@ def handle_app_command(self,
                  ``flask.Response`` object), and the second element
                  is the response status code
         """
+        start_time_ms = time.time() * 1000
+
         # Slightly hacky way to deal with Apple platform
         # smart punctuation messing with argparse.
         cmd_txt = ''.join(map(util.regularize_char, cmd_txt))
         cmd_txt = util.escaped_id_to_id(cmd_txt)
         cmd_txt = util.ios_dash(cmd_txt)
         s = cmd_txt.split(' ', 1)
-        if s[0] == "help" or s[0] is None:
-            logging.info("Help command was called")
+        cmd_name = 'help'
+        if s[0] == 'help' or s[0] is None:
+            logging.info('Help command was called')
             v = self.get_help()
         elif s[0] in self.commands:
             v = self.commands[s[0]].handle(cmd_txt, user)
+            cmd_name = s[0]
         elif is_slack_id(s[0]):
-            logging.info("mention command activated")
-            v = self.commands["mention"].handle(cmd_txt, user)
+            logging.info('mention command activated')
+            v = self.commands['mention'].handle(cmd_txt, user)
+            cmd_name = 'mention'
         else:
             logging.error("app command triggered incorrectly")
             v = self.get_help()
         if isinstance(v[0], str):
             response_data: Any = {'text': v[0]}
         else:
             response_data = v[0]
+
+        # Submit metrics
+        duration_taken_ms = time.time() * 1000 - start_time_ms
+        self.__metrics.submit_cmd_mstime(cmd_name, duration_taken_ms)
```

Comment:
For the command:

```
/rocket team view bob
```

Is `cmd_name` just `team`, or `view`, or `team view bob`? I feel like it should encapsulate the first two (`team view`)

Reply:
`cmd_name` should be `team` and `help` and `user`. with the way things are set up right now, it won't be easy to encapsulate the first two, because some commands (eg mention, help) don't even have that. tho I can add a catch-all for those.

Reply:
I think having the dimension for just `team` would not give us very useful metrics, since that covers such a broad range of functionality. For example, `team refresh` being terrible doesn't mean we need to look at all the team commands

> because some commands (eg mention, help) don't even have that

Ah good point - though we could just leave that empty right?

Reply:
> though we could just leave that empty right?

I... don't know about empty. Maybe make it "not specified".

Reply:
I mean if we join it as a single string:

```
/rocket team view bob -> "team view"
/rocket help -> "help "
/rocket user -> "user"
/rocket user view -> "user view"
```

Reply:
May be too generic. There can be `/rocket user -h` as well.

Reply:
Damn, maybe we can do a "dumb" filter to just not log help events? Ignore if `help` or `-h`